### DISCUSSION TITLE
What on earth happened to 2.0?

### DISCUSSION THREAD

**Discussion by Tequenikality** [YOURLS/YOURLS#3054] [General]
So, I don't mean to "press any buttons" with this post, I know there are some select people on GitHub forums that are sensitive. But the title does speak for itself.

There are references of a YOURLS 2.0 going all the way back to 2012 (9 years ago). So... What happened? Where is it? There are lots of great features that are slated to be in 2.0 - almost like a way of shutting people up when they ask for a new feature by saying it'll be in the mystical 2.0.

There doesn't seem to even be a version of 2.0 anywhere for people to contribute to which would help with the development process. So, what's going on?

**Comment by ozh** [comment]
Hehe. Interesting subject :)

As far as I'm concerned, "2.0" as defined back then was the exact way NOT TO plan development for a tool and a team like YOURLS. It started with a few ideas and quickly ended up with a gigantic PR gathering dozens of features and had become in itself a proper fork of YOURLS. I mean, look at PR 1383 : 315 commits, 450 files changed, started with a simple "hey let's change how the interface looks" and ended up with a whole new software to develop while maintaining the older one. Might have worked for me when I had more free time, but definitely didn't.

Some of the ideas gathered at that time have been and are added to YOURLS, with a more modular approach. There's no strict "roadmap" to be followed.

This said, I admit I like the idea of answering "this will be in 2.0" instead of "meh, I don't have time to work on that feature" 🤣

**Reply by dgw** [reply]
In fairness, it looks like Léo has a private project board with a bunch of v2 features and stuff to work on. @LeoColomb Maybe someday you'll make the 2-do (😏) list public so people can see the progress happening?

I remember that PR, though. Absolutely can't blame anyone for abandoning that unwieldy, unreviewable mess and switching to incremental improvements instead. 😸

**Reply by LeoColomb** [reply]
Ha ha, indeed, and it's funny 'cause I'm focusing on it (again) since 2-3 weeks.
You can see my current wip at https://github.com/YOURLS/yourls-v2 (just transferred, internal only) and my to-do list at https://github.com/orgs/YOURLS/projects/1 (internal too).
@dgw You can play with it, the README part on how to setup the local env in a minute is complete.

I've finished the first piece of the browser interface (i.e. data structures handling and API interactions), this where I am so far (charts are a bit buggy and few other stuff, but you know 🙂.

![localhost_domains_1](https://user-images.githubusercontent.com/846943/132783834-de43da7a-2c11-4cd6-9d4e-50d9fd7e66d2.png)
